density/templates/predict.html

```diff
@@ -9,20 +9,98 @@
 
 <div class="container">
   <div class="col-12"  style="text-align:center;">
-    <h4 style="margin-top: 30px;">Below is the predicted capacity of each building for today: </h4>
+    <h4 style="margin-top: 30px;">Below is the predicted capacity of each building for today.<br><br></h4>
   </div>
 </div>
 
+<div class="container">
+        <div class="searchContainer">
+            <input type="text" class="searchTerm" id="search" onkeyup="searchLocations()" autocomplete= "off">
+            <button type="submit" class="searchButton">
+                <i class="fa fa-search"></i>
+            </button>
+        </div>
+    </div>
+
+    <div class="container">
+        <div class = "btnContainer">
+          <div id="locationBtnContainer" align="left">
+              <input type="radio"  name="location-filter" id="library" value="library" onclick = "searchLocations()"/> <label for="library"> Library </label>
+              <input type="radio"  name="location-filter" id="center" value="center" onclick = "searchLocations()"/> <label for="center"> Student Center </label>
+              <input type="radio"  name="location-filter" id="dining" value="dining" onclick = "searchLocations()"/> <label for="dining"> Dining Hall </label>
+              <button class="clear" onclick="clearFilter('location-filter')">(x)</button>
+          </div>
+
+          <div id="openBtnContainer" align="right">
+              <input type="radio" class="filterTab" name="open-filter" id="open" value="True" onclick = "searchLocations()"/> <label for="open"> Open Now </label>
+              <input type="radio" class="filterTab" name="open-filter" id="closed" value="False" onclick = "searchLocations()"/> <label for="closed">Closed </label>
+              <button class="clear" onclick="clearFilter('open-filter')">(x)</button>
+          </div>
+        </div>
+    </div>
+
 
 <div class="listContainer">
   <ul class="list-group">
     {% for location_name, d in divs.items() %}
-    <li class="list-predict-item"><h4>{{location_name}}</h4>
+    <li class="list-group-item" data-name="{{ location_name }}" 
+                                data-nickname="{{ auxdata[location_name]['nickname'] }}"
+                                data-location-type="{{ auxdata[location_name]['type'] }}"
+                                data-open-now="{{ auxdata[location_name]['open'] }}">
+      <h4>{{location_name}}</h4>
       <div class='graph'>
         {{ d|safe }}
       </div>
     </li>
     {% endfor %}
   </ul>
 </div>
+<script type="text/javascript">
+  function searchLocations() {
```

Comment:
Let's move this into a separate file that's included (or just put it into `static/base.js`) and avoid having it twice.